<!DOCTYPE html>
<!--
Copyright (c) 2013 Samsung Electronics Co., Ltd.

Licensed under the Apache License, Version 2.0 (the License);
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.


Authors:
        Tomasz Paciorek <t.paciorek@samsung.com>

-->
<html>
<head>
<title>ContactManager_removeBatch_errorCallback_TypeMismatch</title>
<meta charset="utf-8"/>
<script type="text/javascript" src="support/unitcommon.js"></script>
</head>

<body>
<div id="log"></div>
<script>
//==== TEST: ContactManager_removeBatch_errorCallback_TypeMismatch
//==== LABEL Check argument ContactManager removeBatch errorCallback conversions exception
//==== PRIORITY: P2
//==== SPEC Tizen Web API:Social:Contact:ContactManager:removeBatch M
//==== SPEC_URL https://developer.tizen.org/help/topic/org.tizen.web.device.apireference/tizen/contact.html
//==== TEST_CRITERIA MC
var t = async_test(document.title), contact1, contact2, addressbook, removeBatchSuccess, i, conversionTable,
    incorrectCallback, exceptionName;
t.step(function () {
    contact1 = new tizen.Contact({
        name: new tizen.ContactName({
            firstName: "Jeffrey",
            lastName: "Hyman"
        })
    });

    contact2 = new tizen.Contact({
        name: new tizen.ContactName({
            firstName: "Elton",
            lastName: "John"
        })
    });
    addressbook = tizen.contact.getDefaultAddressBook();
    addressbook.add(contact1);
    addressbook.add(contact2);
    removeBatchSuccess = t.step_func(function (persons) {
        assert_unreached("Invalid removeBatchSuccess invoked");
    });
    conversionTable = getTypeConversionExceptions("functionObject", true);
    for (i = 0; i < conversionTable.length; i++) {
        incorrectCallback = conversionTable[i][0];
        exceptionName = conversionTable[i][1];
        assert_throws({name: exceptionName},
            function () {
                tizen.contact.removeBatch([contact1.personId, contact2.personId], removeBatchSuccess, incorrectCallback);
            }, exceptionName + " should be thrown - given incorrect.");
    }
    addressbook.remove(contact1.id);
    addressbook.remove(contact2.id);
    t.done();
});

</script>
</body>
</html>
